graylog2-web-interface/src/components/errors/ReportedError.test.jsx

```diff
@@ -0,0 +1,95 @@
+// @flow strict
+import React from 'react';
+import { render, cleanup, waitForElement, waitForElementToBeRemoved } from 'wrappedTestingLibrary';
+
+import suppressConsole from 'helpers/suppressConsole';
+import ErrorsActions from 'actions/errors/ErrorsActions';
+import { createReactError, createUnauthorizedError } from 'logic/errors/ReportedErrors';
+import { FetchError } from 'logic/rest/FetchProvider';
+import ReportedError from './ReportedError';
+
+jest.unmock('logic/rest/FetchProvider');
+jest.mock('react-router', () => ({ withRouter: (x) => x }));
+
+const router = {
+  listen: () => jest.fn(),
+};
+
+describe('ReportedError', () => {
+  afterEach(() => {
+    cleanup();
+  });
+
+  it('registers to router upon mount', () => {
+    const mockRouter = {
+      listen: jest.fn(() => jest.fn()),
+    };
+    render(<ReportedError router={mockRouter}>Hello World!</ReportedError>);
+
+    expect(mockRouter.listen).toHaveBeenCalledTimes(1);
+  });
+
+  it('unregisters from router upon unmount', () => {
+    const unlisten = jest.fn();
+    const mockRouter = {
+      listen: () => unlisten,
+    };
+    const { unmount } = render(<ReportedError router={mockRouter}>Hello World!</ReportedError>);
+
+    unmount();
+
+    expect(unlisten).toHaveBeenCalled();
+  });
+
+
+  it('displays child component if there is no error', () => {
+    const { getByText } = render(<ReportedError router={router}>Hello World!</ReportedError>);
+
+    expect(getByText('Hello World!')).not.toBeNull();
+  });
+
+  it('displays runtime error page when react error got reported', () => {
+    suppressConsole(async () => {
```

Comment:
`suppressConsole` should wrap the smallest possible scope to prevent suppressing valid console output, so maybe we can narrow it down? Why do we need it here anyway, I think there is not exception raised in here?

Reply:
Good point. I removed the not needed `suppressConsole`s and adjusted the scope of the remaining ones, which revealed a broken test.